grype/lib.go

```diff
@@ -31,35 +31,49 @@ func FindVulnerabilities(provider vulnerability.Provider, userImageStr string, s
 
 	matchers := matcher.NewDefaultMatchers(matcher.Config{})
 
-	return FindVulnerabilitiesForPackage(provider, context.Distro, matchers, packages), context, packages, nil
+	return FindVulnerabilitiesForPackage(store, context.Distro, matchers, packages), context, packages, nil
 }
 
-func FindVulnerabilitiesForPackage(provider vulnerability.Provider, d *linux.Release, matchers []matcher.Matcher, packages []pkg.Package) match.Matches {
-	return matcher.FindMatches(provider, d, matchers, packages)
+func FindVulnerabilitiesForPackage(store store.Store, d *linux.Release, matchers []matcher.Matcher, packages []pkg.Package) match.Matches {
+	return matcher.FindMatches(store, d, matchers, packages)
 }
 
-func LoadVulnerabilityDB(cfg db.Config, update bool) (vulnerability.Provider, vulnerability.MetadataProvider, *db.Status, error) {
+func LoadVulnerabilityDB(cfg db.Config, update bool) (*store.Store, error) {
 	dbCurator, err := db.NewCurator(cfg)
 	if err != nil {
-		return nil, nil, nil, err
+		return &store.Store{}, err
```

Comment:
Yeah, so I did this specifically to fix a panic that happens at https://github.com/anchore/grype/blob/finalize-schema-v4/cmd/root.go#L297 when there is no database available.  I guess I should just change `validateDBLoad` to take in the store object rather than just the status?

Or should we not have made the status object part of the store?

Reply:
you could pass the entire store, true. Or, we can refactor the error handling to take place in the caller:
```
			store, err = grype.LoadVulnerabilityDB(appConfig.DB.ToCuratorConfig(), appConfig.DB.AutoUpdate)
			if err != nil {
				errs <- err
				return
			}
			if err = validateDBLoad(store.Status); err != nil {
				errs <- err
				return
			}
```

Reply:
> Or should we not have made the status object part of the store?

Maybe you're right -- that's another approach.

Reply:
I went with taking it back off of the Store object.  It was only used in the validate call from what I could tell